app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -168,9 +168,9 @@ def update_invalid_user
     current_user.increment_second_factor_attempts_count!
   end
 
-  def handle_valid_verification_for_confirmation_context
-    user_session[:authn_at] = Time.zone.now
-    @next_mfa_setup_path = next_setup_path
+  def handle_valid_verification_for_confirmation_context(auth_method:)
+    user_session[:auth_method] = auth_method
+    mark_user_session_authenticated(:valid_2fa_confirmation)
```

Comment:
I'm trying to follow how OTP _previously_ would have set `user_session[TwoFactorAuthenticatable::NEED_AUTHENTICATION]` to `false`, or whether it would have (or even if it needed to?). Just flagging since that will be happening now.